okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -139,32 +140,33 @@ void onStreamSentBytes(int numBytes) {
     onSentBytes(numBytes);
   }
 
+  /**
+   * Must be called with holding the transport lock.
+   */
   public void transportHeadersReceived(List<Header> headers, boolean endOfStream) {
-    synchronized (lock) {
-      if (endOfStream) {
-        transportTrailersReceived(Utils.convertTrailers(headers));
-      } else {
-        transportHeadersReceived(Utils.convertHeaders(headers));
-      }
+    if (endOfStream) {
+      transportTrailersReceived(Utils.convertTrailers(headers));
+    } else {
+      transportHeadersReceived(Utils.convertHeaders(headers));
     }
   }
 
-  /**
-   * We synchronized on "lock" for delivering frames and updating window size, because
-   * the {@link #request(int)} call can be called in other thread for delivering frames.
-   */
   public void transportDataReceived(okio.Buffer frame, boolean endOfStream) {
+    boolean windowExceeded = false;
```

Comment:
I'd generally suggest leaving this unset, because then 1) it is obvious the value here doesn't matter (because there isn't one) and 2) javac will catch if it isn't always set before being used.